python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,433 @@
+                line = "landuse = '${}'".format(os.path.join(USRDAT_DIR, fluse_out))
+                self.write_to_file(line, nl_clm)
+
+    def create_surfdata_at_point(self, indir, file, user_mods_dir):
+        """
+        Create surface data file at a single point.
+        """
+        logger.info("----------------------------------------------------------------------")
+        logger.info(
+            "Creating surface dataset file at %s, %s", self.plon.__str__(), self.plat.__str__())
+
+        # specify file
+        fsurf_in = os.path.join(indir, file)
+        fsurf_out = add_tag_to_filename(fsurf_in, self.tag)
+        logger.info("fsurf_in:  %s", fsurf_in)
+        logger.info("fsurf_out: %s", os.path.join(self.output_dir, fsurf_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(fsurf_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat")
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"]).copy(deep=True)
+
+        # update the plon and plat to match the surface data
+        # we do this so that if we create user_mods the PTS_LON and PTS_LAT in CIME match
+        # the surface data coordinates - which is required
+        self.plat = f_out.coords["lsmlat"].values[0]
+        self.plon = f_out.coords["lsmlon"].values[0]
+
+        # modify surface data properties
+        if self.overwrite_single_pft:
+            f_out["PCT_NAT_PFT"][:, :, :] = 0
+            if self.dominant_pft < 16:
+                f_out['PCT_NAT_PFT'][:, :, self.dominant_pft] = 100
+        if self.zero_nonveg_landunits:
+            f_out["PCT_NATVEG"][:, :] = 100
+            f_out["PCT_CROP"][:, :] = 0
+            f_out["PCT_LAKE"][:, :] = 0.0
+            f_out["PCT_WETLAND"][:, :] = 0.0
+            f_out["PCT_URBAN"][:, :, ] = 0.0
+            f_out["PCT_GLACIER"][:, :] = 0.0
+        if self.uniform_snowpack:
+            f_out["STD_ELEV"][:, :] = 20.0
+        if not self.saturation_excess:
+            f_out["FMAX"][:, :] = 0.0
+
+        # specify dimension order
+        f_out = f_out.transpose(
+            u"time",
+            u"cft",
+            u"lsmpft",
+            u"natpft",
+            u"nglcec",
+            u"nglcecp1",
+            u"nlevsoi",
+            u"nlevurb",
+            u"numrad",
+            u"numurbl",
+            "lsmlat",
+            "lsmlon",
+        )
+
+        # update lsmlat and lsmlon to match site specific instead of the nearest point
+        f_out['lsmlon'] = np.atleast_1d(self.plon)
+        f_out['lsmlat'] = np.atleast_1d(self.plat)
+        f_out['LATIXY'][:, :] = self.plat
+        f_out['LONGXY'][:, :] = self.plon
+
+        # update attributes
+        self.update_metadata(f_out)
+        f_out.attrs["Created_from"] = fsurf_in
+        del f_out.attrs["History_Log"]
```

Comment:
This line should be remove because we are already removing it in update_metadata.

Reply:
removed.